src/AdoNetCore.AseClient/Internal/InternalConnection.cs

```diff
@@ -662,18 +662,23 @@ private string MakeCommand(string commandText, CommandBehavior behavior, bool na
             return result;
         }
 
-        private IToken[] BuildParameterTokens(AseParameterCollection parameters)
+        private IToken[] BuildParameterTokens(AseCommand command)
         {
             var formatItems = new List<FormatItem>();
             var parameterItems = new List<ParametersToken.Parameter>();
 
-            foreach (var parameter in parameters.SendableParameters)
+            foreach (var parameter in command.Parameters.SendableParameters)
             {
-                var formatItem = FormatItem.CreateForParameter(parameter, _environment);
-                formatItems.Add(formatItem);
+                var parameterName = parameter.ParameterName ?? command.Parameters.IndexOf(parameter).ToString();
```

Comment:
I still don't understand why command has a `FormatItem` at all?

Why not just:
```
var formatItem = FormatItem.CreateForParameter(parameter, _environment, command.CommandType);
formatItems.Add(formatItem);
parameterItems.Add(new ParametersToken.Parameter
{
    Format = formatItem
    Value = parameter.SendableValue
});
```

Reply:
If you have a TEXT type column but the the length of the text value doesn't qualify it to be a _TEXT_ .
This is Ok for the most part but say if you have to chunk up the text value into smaller chunk, the last chunk might fail upon insert because of the length might not qualify it to be in a _TEXT_ column.
By have a cache version of the FormatItem in the command it guarantee that last chunk to have the metadata as the previous chunks hence insert/update would succeed.